pallets/automation-time/src/lib.rs

```diff
@@ -765,33 +827,83 @@ pub mod pallet {
 			<T as Config>::WeightInfo::run_native_transfer_task()
 		}
 
+		/// Decrements task executions left.
+		/// If task is complete then removes task. If task not complete update task map.
+		/// A task has been completed if executions left equals 0.
+		fn decrement_task_and_remove_if_complete(task_id: T::Hash, mut task: Task<T>) {
```

Comment:
`task_id` is not stored in the task object.  I could call `let task_id = Self::generate_task_id(owner_id.clone(), provided_id.clone());` to get the `task_id` and that would allow to not pass it in.
`